File: rust/src/models/one_time_pricing.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct OneTimePricing {
    pub quantity: i32,

    pub unit_price: rust_decimal::Decimal,
}

impl OneTimePricing {
    pub fn new(quantity: i32, unit_price: rust_decimal::Decimal) -> Self {
        Self {
            quantity,
            unit_price,
        }
    }
}

File: typescript/src/models/oneTimePricing.ts
// this file is @generated

export interface OneTimePricing {
  quantity: number;

  unitPrice: string;
}

export const OneTimePricingSerializer = {
  _fromJsonObject(object: any): OneTimePricing {
    return {
      quantity: object["quantity"],
      unitPrice: object["unit_price"],
    };
  },

  _toJsonObject(self: OneTimePricing): any {
    return {
      quantity: self.quantity,
      unit_price: self.unitPrice,
    };
  },
};
